firmware/sys/network/cpumac.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * @brief  cpuid address
+ *
+ * @author  RocioRojas <rociorojas391@gmail.com>
+ */
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "cpumac.h"
+
+void cpumac(char *input, char *output) {
+    int loop = 0;
+    int i = 0;
+    CPUID(input);
+    for (int k = 0; k < (int)CPUID_LEN; k++) {
+        sprintf((char *)(output + i), "%02x", input[loop]);
+        loop += 1;
+        i += 2;
+    }
+    // insert NULL at the end of the output string
+    output[i++] = '\0';
+}
+
+void mac_ipv6(char *mac) {
+
+    char address[CPUID_LEN] = {0};
+    char addr_hex[CPUID_LEN * 2] = {0};
+    cpumac(address, addr_hex);
+    strcpy(mac, CONFIG_HEADER_ADDRESS_MAC);
+    strncat(mac, addr_hex, 4);
+    strcat(mac, ":");
+    strncat(mac, addr_hex + 4, 4);
+}
```

Comment:
- the function's name is not descriptive
- a string cant be a `ipv6`, use the correct data type